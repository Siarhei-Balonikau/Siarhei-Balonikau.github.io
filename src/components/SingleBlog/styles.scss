.single-blog {
  &__title {
    text-align: center;
    font-size: 22px;
    font-weight: 700;
    padding-bottom: 10px;
  }
  
  &__text {
    font-size: 16px;
    padding-bottom: 30px;
  }
  
  &__author {
    font-size: 12px;
  }
  
  &__link {
    display: block;
    border: 1px solid rgba(black, 0.3);
    transition: 0.3s;
    text-align: center;
    margin-bottom: 30px;
    color: black;
    text-decoration: none;
    padding: 15px 0;
      
    &:hover {
      background-color: black;
      color: white;
    }
  }
}