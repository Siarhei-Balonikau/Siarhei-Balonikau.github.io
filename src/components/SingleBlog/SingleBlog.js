import React from 'react';
import PropTypes from 'prop-types';
import { Router, Route, Link } from "react-router-dom";
import './styles.scss';


const SingleBlog = props => {
  const { blog } = props;

  return (
    <div className="single-blog">
      <Link className="single-blog__link" to={`/`}>Show all blogs</Link>
      
      <div className="single-blog__title">
        {blog.title}
      </div>
        
      <div className="single-blog__text">
        {blog.text}
      </div>
        
      <div className="single-blog__author">
        {blog.author}
      </div>
    </div>
  );
};

SingleBlog.propTypes = {
  blog: PropTypes.object.isRequired
};

export default SingleBlog;