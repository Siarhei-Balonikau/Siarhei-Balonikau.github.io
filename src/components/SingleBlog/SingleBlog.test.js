import React, { Component } from 'react';
import { shallow } from 'enzyme';
import SingleBlog from './SingleBlog'; 
import toJson from 'enzyme-to-json'

describe('subscribe component', () => {
  let wrapper, props;

  beforeEach(() => {
    props = {
      blog: {
        title: 'test',
        text: 'test',
        author: 'test'
      }
    };
  
    wrapper = shallow(<SingleBlog {...props} />);
  });
  
  it('should render component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});