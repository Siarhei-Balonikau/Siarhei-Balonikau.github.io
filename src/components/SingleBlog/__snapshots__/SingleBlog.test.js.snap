// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subscribe component should render component 1`] = `
<div
  className="single-blog"
>
  <Link
    className="single-blog__link"
    replace={false}
    to="/"
  >
    Show all blogs
  </Link>
  <div
    className="single-blog__title"
  >
    test
  </div>
  <div
    className="single-blog__text"
  >
    test
  </div>
  <div
    className="single-blog__author"
  >
    test
  </div>
</div>
`;
